### PR TITLE
9615 docket search capability

### PR DESCRIPTION
**Pull request by rgoshen**

### CODE REVIEW

File: web-client/integration-tests/journey/petitionerSearchesForUnassociatedSealedCase.js [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+const caseDetailHelper = withAppContextDecorator(caseDetailHelperComputed);
+
+export const petitionerSearchesForUnassociatedSealedCase = cerebralTest => {
+  return it('petitioner searches for unassociated sealed case', async () => {
+    // cerebralTest.setState('caseDetail', {});
+    const sealedSeedCaseDocketNumber = '105-20';
+
+    await cerebralTest.runSequence('gotoDashboardSequence');
+
+    // cerebralTest.setState('header.searchTerm', sealedSeedCaseDocketNumber);
```
Comment (mmarcotte) [inline web-client/integration-tests/journey/petitionerSearchesForUnassociatedSealedCase.js]: hm, i think these comments are lingering from testing and/or still helpful?

Comment (rgoshen) [reply web-client/integration-tests/journey/petitionerSearchesForUnassociatedSealedCase.js]: removed-thanks

File: web-client/src/presenter/computeds/dashboardExternalHelper.test.js [hunk 2]
```diff
@@ -36,50 +36,65 @@ describe('petitioner dashboard helper', () => {
...
     });
-    expect(result.showCaseList).toEqual(true);
-    expect(result.showWhatToExpect).toEqual(false);
-    expect(result.showCaseSearch).toEqual(true);
+
+    expect(result.showStartButton).toEqual(false);
   });
 
-  it('hides case search if defined user does not have privatePractitioner or irsPractitioner role', () => {
+  it('should set the showStartButton flag as true when the user role is a private practitioner or petitioner', () => {
```
Comment (mmarcotte) [inline web-client/src/presenter/computeds/dashboardExternalHelper.test.js:86]: hm, it seems like if you want to be true to this `it('...')` statement, you'd want to run this test twice for `privatePractitioner` and `petitioner`. Perhaps add these roles to an array and loop through them?

File: web-client/src/presenter/computeds/dashboardExternalHelper.test.js [hunk 3]
```diff
@@ -36,50 +36,65 @@ describe('petitioner dashboard helper', () => {
     });
     expect(result.showCaseList).toEqual(true);
     expect(result.showWhatToExpect).toEqual(false);
-    expect(result.showCaseSearch).toEqual(false);
   });
-  it('shows case search if defined user has privatePractitioner role', () => {
+
+  it('should keep the showFileACase flag as false when the user role is not a private practitioner', () => {
```
Comment (mmarcotte) [inline web-client/src/presenter/computeds/dashboardExternalHelper.test.js]: Hm, for this `it(...)` statement, might me want to explicit for being a `petitioner` or loop through `petitioner` and  `irsrPractitioner`?

File: web-client/src/presenter/computeds/dashboardExternalHelper.test.js [hunk 4]
```diff
@@ -36,50 +36,65 @@ describe('petitioner dashboard helper', () => {
...
     });
-    expect(result.showCaseList).toEqual(true);
-    expect(result.showWhatToExpect).toEqual(false);
-    expect(result.showCaseSearch).toEqual(true);
+
+    expect(result.showFileACase).toEqual(true);
   });
 
-  it('shows case search if defined user has irsPractitioner role', () => {
+  it('should keep the showStartButton flag as false when the user role is not a private practitioner or petitioner', () => {
```
Comment (mmarcotte) [inline web-client/src/presenter/computeds/dashboardExternalHelper.test.js]: Perhaps to keep things less confusing, explicitly state `irsPractitioner`

File: web-client/src/presenter/computeds/dashboardExternalHelper.test.js (hunk 3 above)
Comment (rgoshen) [reply web-client/src/presenter/computeds/dashboardExternalHelper.test.js]: add an roles array and looped through it

File: web-client/src/presenter/computeds/dashboardExternalHelper.test.js (hunk 4 above)
Comment (rgoshen) [reply web-client/src/presenter/computeds/dashboardExternalHelper.test.js]: done

File: web-client/src/presenter/computeds/dashboardExternalHelper.test.js (hunk 3 above)
Comment (rgoshen) [reply web-client/src/presenter/computeds/dashboardExternalHelper.test.js:41]: changed it block wording to "...when the user role is petitioner"

File: web-client/src/presenter/computeds/dashboardExternalHelper.test.js (hunk 2 above)
Comment (rgoshen) [reply web-client/src/presenter/computeds/dashboardExternalHelper.test.js:86]: changed it block wording to "...when the user role is petitioner"

**Review by mmarcotte — APPROVED** [review]

**Pull request merged**